customcollection.go

```diff
@@ -0,0 +1,96 @@
+	Count(interface{}) (int, error)
+	Get(int, interface{}) (*CustomCollection, error)
+	Create(CustomCollection) (*CustomCollection, error)
+	Update(int, CustomCollection) (*CustomCollection, error)
+	Delete(int) error
+}
+
+// CustomCollectionServiceOp handles communication with the custom collection
+// related methods of the Shopify API.
+type CustomCollectionServiceOp struct {
+	client *Client
+}
+
+// CustomCollection represents a Shopify custom collection.
+type CustomCollection struct {
+	ID             int    `json:"id"`
+	Handle         string `json:"handle"`
+	Title          string `json:"title"`
+	UpdatedAt      string `json:"updated_at"`
+	BodyHTML       string `json:"body_html"`
+	SortOrder      string `json:"sort_order"`
+	TemplateSuffix string `json:"template_suffix"`
+	Image          Image  `json:"image"`
+	Published      bool   `json:"published"`
+	PublishedAt    string `json:"published_at"`
+	PublishedScope string `json:"published_scope"`
+}
+
+// CustomCollectionResource represents the result form the custom_collections/X.json endpoint
+type CustomCollectionResource struct {
+	Collection *CustomCollection `json:"custom_collection"`
+}
+
+// CustomCollectionsResource represents the result from the custom_collections.json endpoint
+type CustomCollectionsResource struct {
+	Collections []CustomCollection `json:"custom_collections"`
+}
+
+// List custom collections
+func (s *CustomCollectionServiceOp) List(options interface{}) ([]CustomCollection, error) {
+	path := fmt.Sprintf("%s.json", customCollectionsBasePath)
+	resource := new(CustomCollectionsResource)
+	err := s.client.Get(path, resource, options)
+	return resource.Collections, err
+}
+
+// Count custom collections
+func (s *CustomCollectionServiceOp) Count(options interface{}) (int, error) {
+	path := fmt.Sprintf("%s/count.json", customCollectionsBasePath)
+	return s.client.Count(path, options)
+}
+
+// Get individual custom collection
+func (s *CustomCollectionServiceOp) Get(collectionID int, options interface{}) (*CustomCollection, error) {
+	path := fmt.Sprintf("%s/%d.json", customCollectionsBasePath, collectionID)
+	resource := new(CustomCollectionResource)
+	err := s.client.Get(path, resource, options)
+	return resource.Collection, err
+}
+
+// Create a new custom collection
+// See Image for the details of the Image creation for a collection.
+func (s *CustomCollectionServiceOp) Create(collection CustomCollection) (*CustomCollection, error) {
+	path := fmt.Sprintf("%s.json", customCollectionsBasePath)
+	wrappedData := CustomCollectionResource{Collection: &collection}
+	resource := new(CustomCollectionResource)
+	err := s.client.Post(path, wrappedData, resource)
+	return resource.Collection, err
+}
+
+// Update an existing custom collection
+func (s *CustomCollectionServiceOp) Update(collectionID int, collection CustomCollection) (*CustomCollection, error) {
+	path := fmt.Sprintf("%s/%d.json", customCollectionsBasePath, collectionID)
```

Comment:
Could you change the signature of the `Update` function so it takes only the `CustomCollection` to update as parameter and then uses the `ID` field? See e.g. the [`Product`](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/product.go#L101-L102) and [`Variant`](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/variant.go#L100-L101) `Update` functions.